pkg/controller/stateful_set_control.go

```diff
@@ -98,6 +100,7 @@ func (c *realStatefulSetControl) UpdateStatefulSet(controller runtime.Object, se
 			// make a copy so we don't mutate the shared cache
 			set = updated.DeepCopy()
 			set.Spec = *setSpec
+			_ = helper.SetDeleteSlots(set, deleteSlots)
```

Comment:
I think we should reset all the labels and annotations of the sts. Refer to https://github.com/pingcap/tidb-operator/blob/v1.3.2/pkg/manager/utils/statefulset.go#L128-L129.